libxrdp/xrdp_sec.c

```diff
@@ -778,20 +778,20 @@ xrdp_sec_process_logon_info(struct xrdp_sec *self, struct stream *s)
         return 1;
     }
 
-    if (unicode_in(s, len_domain, self->rdp_layer->client_info.domain, 255) != 0)
+    if (unicode_in(s, len_domain, 512, self->rdp_layer->client_info.domain, 255) != 0)
```

Comment:
It should be clear whether the number comes from the protocol (in which case it would be defined next to other protocol specific constants) or it's an arbitrary limit (in which case it can be defined in the same file as the implementation).